### PR TITLE
feat: add vulkan images

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

libkrun provider has been added to podman and soon we should have a way to start an inference server that is able to use gpu on a mac. This PR adds the creation of vulkan images so that, eventually, we can use them to start an inference server with gpu capabilities.

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

N/A

### How to test this PR?

1. check that the image build correctly

### CODE REVIEW

File: chat/vulkan/amd64/Containerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM registry.access.redhat.com/ubi9/python-311:1-66
```
Comment (benoitf) [inline chat/vulkan/amd64/Containerfile:1]: missing copyrights ?

File: chat/vulkan/amd64/Containerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+USER 0
+RUN dnf install -y python3-dnf-plugin-versionlock
+RUN dnf install -y mesa-vulkan-drivers-23.3.3-1.el9.x86_64
+RUN dnf versionlock mesa-vulkan-drivers-23.3.3-1.el9.x86_64
+RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm
+RUN dnf install -y git cmake ninja-build gcc gcc-c++ 
+RUN dnf copr enable -y ligenix/enterprise-sandbox epel-9-x86_64
+RUN dnf install -y vulkan-headers vulkan-tools
```
Comment (benoitf) [inline chat/vulkan/amd64/Containerfile:9]: as it looks like it's not a multi-stage build, you should mergre layers and cleanup .rpm files / rpm cache at the end to reduce the size of the image

or maybe better, use the default image and copy the custom rootfs so we've a scratch/smaller image (this is how ubi micro is done)

File: chat/vulkan/arm64/Containerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
+FROM registry.access.redhat.com/ubi9/python-311:1-66
```
Comment (benoitf) [inline chat/vulkan/arm64/Containerfile:1]: headers

**Review by jeffmaury — COMMENTED** [review]
The one from ai-lab-recipes uses library krunkit specific ?

**Comment by lstocchi** [comment]
> The one from ai-lab-recipes uses library krunkit specific ?

It is primarily intended to be used on macOS with podman and krunkit but Sergio also said that it can be used on Linux with a VMM exposing a Venus-capable virtio-gpu.

**Comment by jeffmaury** [comment]
> > The one from ai-lab-recipes uses library krunkit specific ?
> 
> It is primarily intended to be used on macOS with podman and krunkit but Sergio also said that it can be used on Linux with a VMM exposing a Venus-capable virtio-gpu.

Did you check that performances with this images are the same compared to the benchmarks we did on MacOS/krunkit ?

**Comment by lstocchi** [comment]
> Did you check that performances with this images are the same compared to the benchmarks we did on MacOS/krunkit ?

What do you mean? Didn't we use the image generated by the ai-lab-recipes team for the benchmark? I just copy/paste the containerfile so the result should be the same, no?

**Comment by jeffmaury** [comment]
> > Did you check that performances with this images are the same compared to the benchmarks we did on MacOS/krunkit ?
> 
> What do you mean? Didn't we use the image generated by the ai-lab-recipes team for the benchmark? I just copy/paste the containerfile so the result should be the same, no?

Does not seem to me as the ai-lab-recipe one contains libkrun specific libraries: see https://github.com/containers/ai-lab-recipes/blob/624155c699b7c2abc3c5878fba21ad848b23af25/model_servers/llamacpp_python/vulkan/arm64/Containerfile#L6

File: .github/workflows/build-publish.yaml [hunk 4]
```diff
@@ -28,6 +28,19 @@ jobs:
...
     name: Build image
     runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        include:
+          - containerfile: "./chat/base/Containerfile"
+            build-image-name: "ai-lab-playground-chat"
+            archs: amd64, arm64
+          - containerfile: "./chat/vulkan/amd64/Containerfile"
+            build-image-name: "ai-lab-playground-chat-vulkan-amd64"
```
Comment (jeffmaury) [inline .github/workflows/build-publish.yaml:38]: The image name should not contain the arch so this should be handled through a manifest

**Comment by lstocchi** [comment]
> Does not seem to me as the ai-lab-recipe one contains libkrun specific libraries: see https://github.com/containers/ai-lab-recipes/blob/624155c699b7c2abc3c5878fba21ad848b23af25/model_servers/llamacpp_python/vulkan/arm64/Containerfile#L6

-> https://github.com/containers/podman-desktop-extension-ai-lab-playground-images/pull/22/files#diff-69e31da53e7e524501214e0e65d027964ba095c665858c15aa178fa24379d445R6

File: .github/workflows/build-publish.yaml (hunk 4 above)
Comment (lstocchi) [reply .github/workflows/build-publish.yaml:38]: there are two versions of the same image. One for amd and one for arm. We should use the one we need based on the system

Comment (jeffmaury) [reply .github/workflows/build-publish.yaml:38]: Yes but we should select based on the arch not the name just like the base image (CPU)

Comment (lstocchi) [reply .github/workflows/build-publish.yaml:38]: Removed the arch from the name. is that enough?

File: chat/vulkan/amd64/Containerfile (hunk 2 above)
Comment (lstocchi) [reply chat/vulkan/amd64/Containerfile]: I've been trying to copy what they did on ubi-micro but i cannot make it work with installroot. There is something else to set up but i was not able to find it. 
I went to the good old way to check which folders to copy and i added the commands. If you know how to make the custom rootfs works, i'm all ears

File: .github/workflows/build-publish.yaml (hunk 4 above)
Comment (lstocchi) [reply .github/workflows/build-publish.yaml:38]: Ops, updated the other file :face-palm:

Comment (lstocchi) [reply .github/workflows/build-publish.yaml]: Updated

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -32,8 +32,13 @@ env:
 
 jobs:
 
-  build:
+  tag:
```
Comment (jeffmaury) [inline .github/workflows/release.yaml:35]: Why do you need this extra job ?

Comment (lstocchi) [reply .github/workflows/release.yaml]: I thought that separating it in various jobs was more clean. Updated to merge all jobs into one as before

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**